### PR TITLE
Use Roman Numeral Converter to create api endpoint

### PR DESCRIPTION
**Pull request by palledorous**

### CODE REVIEW

**Comment by mikeknep** [comment]
I use this CLI on a locked-down machine that doesn't have `jq` or permissions to fire up a server. I really liked using the CLI to convert roman numerals! It saved me hours! Can you continue to support the CLI alongside this new HTTP option?

**Comment by palledorous** [comment]
@mikeknep that's a good suggestion. I don't see any reason why not!

File: lib/converter.rb [hunk 1]
```diff
@@ -28,7 +30,8 @@ def convert_roman_to_arabic(input)
                 result += current_value
                 i += 1
             end
+            converted_value = { :value => result }
         end
-        result
+        JSON.generate(converted_value)
```
Comment (pedmiston) [inline lib/converter.rb]: Why does the `convert_roman_to_arabic` need to change its interface in order to accommodate the API server? I feel that the server should be the thing that converts responses to JSON.

File: spec/converter_spec.rb [hunk 2]
```diff
@@ -17,7 +17,10 @@
 
     cases.each do |numeral, value|
         it 'returns the proper value when passed a roman numeral string' do
-            expect(Converter.new.convert_roman_to_arabic(numeral)).to eq(value)
+
+            result = JSON.parse(Converter.new.convert_roman_to_arabic(numeral))
```
Comment (pedmiston) [inline spec/converter_spec.rb]: Related to my comment above, this test shouldn't have to change to accommodate the new API endpoint.

File: app/application.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+        if request.path.match(/convert/)
+            numeral = request.params["roman"]
+
+            converted_numeral = Converter.new.convert_roman_to_arabic(numeral)
+            result            = JSON.parse(converted_numeral)
+
+            response.status   = 200
+            response.write "\n"
+            response.write result["value"]
```
Comment (pedmiston) [inline app/application.rb]: Let's have the endpoint return a valid JSON string in the response body. Something like:

```
curl localhost:8000/convert&roman=IV
# {
#   "value": 4
# }
```

File: spec/application_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        get '/convert?roman=XXX'
+
+        expect(last_response.status).to eq(200)
+        expect(last_response.body).to include("30")
+    end
+
+    it 'returns a 404 when requested a path that does not exist' do
+        get '/whoops'
+
+        expect(last_response.status).to eq(404)
```
Comment (pedmiston) [inline spec/application_spec.rb]: I'm glad you added this test! Very nice!

File: lib/converter.rb (hunk 1 above)
Comment (mikeknep) [reply lib/converter.rb]: This is what I was sneakily getting at with my goofy comment. Consider the difference between some core logic and how the result of that logic is _presented_ to some user.

**Pull request merged**